system/jlib/jlog.cpp

```diff
@@ -1003,13 +1003,21 @@ void FileLogMsgHandlerXML::addToPTree(IPropertyTree * tree) const
 
 // RollingFileLogMsgHandler
 
-RollingFileLogMsgHandler::RollingFileLogMsgHandler(const char * _filebase, const char * _fileextn, unsigned _fields, bool _append, bool _flushes, const char *initialName, const char *_alias, bool daily)
-  : handle(0), messageFields(_fields), alias(_alias), filebase(_filebase), fileextn(_fileextn), append(_append), flushes(_flushes)
+RollingFileLogMsgHandler::RollingFileLogMsgHandler(const char * _filebase, const char * _fileextn, unsigned _fields, bool _append, bool _flushes, const char *initialName, const char *_alias, bool daily, long _maxLogFileSize)
+  : handle(0), messageFields(_fields), alias(_alias), filebase(_filebase), fileextn(_fileextn), append(_append), flushes(_flushes), maxLogFileSize(_maxLogFileSize)
 {
+    if (_maxLogFileSize)
+    {
+        if (_maxLogFileSize < 10000)                 // Setting the cap too low, doesn't work well
```

Comment:
It would be better if the constants 10000, 150, 160 had symbolic names e.g. minLogFileLimit, logLineEstimate